gcp/api/server.py

```diff
@@ -111,29 +111,45 @@ def QueryAffectedBatch(self, request, context):
   @ndb_context
   def DetermineVersion(self, request, context):
     """Determine the version of the provided hashes."""
-    if not request.query.name:
-      context.abort(grpc.StatusCode.NOT_IMPLEMENTED,
-                    'Querying only by file hash is not implemented yet.')
-      return None
-    return get_version_by_name(request.query).result()
+    return determine_version(request.query, context).result()
 
 
 @ndb.tasklet
-def get_version_by_name(
-    version_query: osv_service_v1_pb2.VersionQuery) -> ndb.Future:
-  """Identifies the version based on the provided name."""
-  query = osv.RepoIndex.query(osv.RepoIndex.name == version_query.name)
-  it = query.iter()
+def determine_version(version_query: osv_service_v1_pb2.VersionQuery,
+                      context: grpc.ServicerContext) -> ndb.Future:
+  """Identify fitting commits based on a subset of hashes"""
+  hashes = []
+  for i in range(min(20, len(version_query.file_hashes))):
+    hashes.append(version_query.file_hashes[i].hash)
+  idx_keys_set = set()
+  for h in hashes:
+    query = osv.RepoIndexResult.query(
+        osv.RepoIndexResult.file_results.hash == h)
```

Comment:
We can likely structure this in a better way to take advantage of the async querying such that we can actually run these queries in parallel.  

Right now we're doing this sequentially because we're waiting for each query to complete before we proceed to the next hash.

Reply:
PTAL